perllib/FixMyStreet/App/Controller/Dashboard.pm

```diff
@@ -126,8 +126,32 @@ sub index : Path : Args(0) {
         $c->stash->{contacts} = [ $c->stash->{contacts}->all ];
         $c->forward('/report/stash_category_groups', [ $c->stash->{contacts} ]);
 
+        my %group_names = map { $_->{name} => 1 } @{$c->stash->{category_groups}};
+
         # See if we've had anything from the body dropdowns
-        $c->stash->{category} = $c->get_param('category');
+        $c->stash->{category} = [ $c->get_param_list('category') ];
+        my @remove_from_display;
+        if (grep { $_ =~ /^group-/ } @{ $c->stash->{category} }) {
```

Comment:
Unless I've missed an oddity in what this is trying to do, would something like this construction have the same effect as what you're after?

```
my %group_names = map { $_->{name} => $_->{categories} } @{$c->stash->{category_groups}};
foreach (@{$c->stash->{category}}) {
    next unless /^group-(.*)/;
    for my $contact (@{$group_names{$1}}) {
        push @{ $c->stash->{category} }, $contact->category;
        push @remove_from_display, $contact->category;
    }
}
```